x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EsQueryExec.java

```diff
@@ -56,8 +59,19 @@ public class EsQueryExec extends LeafExec implements EstimatesRowSize {
      */
     private final Integer estimatedRowSize;
 
-    public record FieldSort(FieldAttribute field, Order.OrderDirection direction, Order.NullsPosition nulls) implements Writeable {
-        public FieldSortBuilder fieldSortBuilder() {
+    public interface Sort {
+        SortBuilder<?> sortBuilder();
+
+        Order.OrderDirection direction();
+
+        FieldAttribute field();
+
+        void writeTo(StreamOutput out) throws IOException;
```

Comment:
I wonder if we really need to serialize this? EsQueryExec as a whole might never be serialized.

Reply:
I wanted to leave the complete removal of serialization support from EsQueryExec to a separate PR; and so I kept serialization in but made it all dummy serialization. However, your point is valid that some of the remaining code is no longer required since I made the serialization of the Sort all dummy serialization (ie. empty lists). I still need to de-serialize, because we can be receiving from an old node, but at least the `writeTo` methods can go.

Reply:
Done.